File: apps/admin/src/services/commentService.ts
import api from './api';
import { R, PageResult } from '@/types';

export enum CommentStatus {
  PENDING = 'PENDING',
  APPROVED = 'APPROVED',
  REJECTED = 'REJECTED',
  SPAM = 'SPAM',
  DELETED = 'DELETED'
}

export interface Comment {
  id: number;
  nickname: string;
  email: string;
  website?: string;
  avatar?: string;
  content: string;
  status: CommentStatus;
  ip?: string;
  userAgent?: string;
  isAdmin: boolean;
  likeCount: number;
  createdAt: string;
  updatedAt: string;
  post?: {
    id: number;
    title: string;
    slug: string;
  };
  parent?: {
    id: number;
    nickname: string;
  };
}

class CommentService {
  async listAll(status?: CommentStatus, pageNum: number = 1, pageSize: number = 10): Promise<R<PageResult<Comment>>> {
    const params = { status, pageNum, pageSize };
    return api.get<R<PageResult<Comment>>>('/v1/admin/comments', { params });
  }

  async getById(id: number): Promise<R<Comment>> {
    return api.get<R<Comment>>(`/v1/admin/comments/${id}`);
  }

  async approve(id: number): Promise<R<Comment>> {
    return api.patch<R<Comment>>(`/v1/admin/comments/${id}/approve`);
  }

  async reject(id: number): Promise<R<Comment>> {
    return api.patch<R<Comment>>(`/v1/admin/comments/${id}/reject`);
  }

  async markAsSpam(id: number): Promise<R<Comment>> {
    return api.patch<R<Comment>>(`/v1/admin/comments/${id}/spam`);
  }

  async restore(id: number): Promise<R<Comment>> {
    return api.patch<R<Comment>>(`/v1/admin/comments/${id}/restore`);
  }

  async delete(id: number): Promise<R<void>> {
    return api.delete<R<void>>(`/v1/admin/comments/${id}`);
  }

  async permanentDelete(id: number): Promise<R<void>> {
    return api.delete<R<void>>(`/v1/admin/comments/${id}/permanent`);
  }

  async batchDelete(ids: number[]): Promise<R<void>> {
    return api.delete<R<void>>('/v1/admin/comments/batch', { data: ids });
  }

  async batchPermanentDelete(ids: number[]): Promise<R<void>> {
    return api.delete<R<void>>('/v1/admin/comments/batch/permanent', { data: ids });
  }

  async batchApprove(ids: number[]): Promise<R<void>> {
    return api.patch<R<void>>('/v1/admin/comments/batch/approve', { data: ids });
  }
}

export const commentService = new CommentService();
export default commentService;

File: apps/server/aetherblog-service/blog-service/src/main/java/com/aetherblog/blog/service/CommentService.java
package com.aetherblog.blog.service;

import com.aetherblog.blog.entity.Comment;
import com.aetherblog.blog.entity.Comment.CommentStatus;
import com.aetherblog.common.core.domain.PageResult;

import java.util.List;

/**
 * 评论服务接口
 * 
 * @ref §3.2-11 - 评论管理模块
 */
public interface CommentService {

    /**
     * 获取待审核评论列表
     */
    PageResult<Comment> listPending(int pageNum, int pageSize);

    /**
     * 获取所有评论列表（管理后台）
     */
    PageResult<Comment> listAll(CommentStatus status, int pageNum, int pageSize);

    /**
     * 获取文章的评论列表（已审核通过）
     */
    PageResult<Comment> listByPost(Long postId, int pageNum, int pageSize);

    /**
     * 获取评论详情
     */
    Comment getById(Long id);

    /**
     * 创建评论（前台访客提交）
     */
    Comment create(Long postId, Comment comment);

    /**
     * 审核通过
     */
    Comment approve(Long id);

    /**
     * 审核拒绝
     */
    Comment reject(Long id);

    /**
     * 标记为垃圾评论
     */
    Comment markAsSpam(Long id);

    /**
     * 还原评论
     */
    Comment restore(Long id);

    /**
     * 软删除评论（移至回收站）
     */
    void delete(Long id);

    /**
     * 彻底删除评论
     */
    void permanentDelete(Long id);

    /**
     * 批量删除评论（软删除）
     */
    void batchDelete(List<Long> ids);

    /**
     * 批量彻底删除评论
     */
    void batchPermanentDelete(List<Long> ids);

    /**
     * 批量审核通过
     */
    void batchApprove(List<Long> ids);

    /**
     * 获取文章评论数
     */
    long countByPost(Long postId);

    /**
     * 回复评论
     */
    Comment reply(Long parentId, Comment comment);
}
